src/App.tsx

```diff
@@ -1,61 +1,105 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { Todos, TodoWithUser, User } from './types';
+
+const todosWithUsers = (todos: Todos[], users: User[]): TodoWithUser[] => {
+  return todos.map(todo => {
+    const match = users.find(user => user.id === todo.userId);
+
+    return {
+      ...todo,
+      name: match?.name || '',
+      username: match?.username || '',
+      email: match?.email || '',
+    };
+  });
+};
 
 export const App = () => {
+  const [title, setTitle] = useState<string>('');
+  const [onSubmit, setOnSubmit] = useState<boolean>(false);
+  const [selectedUser, setSelectedUser] = useState<string>('0');
+  const [todos, setTodos] = useState<TodoWithUser[]>(
+    todosWithUsers(todosFromServer, usersFromServer),
+  );
+
+  const handleOnSubmit = (value: React.FormEvent) => {
+    value.preventDefault();
+
+    if (!title || selectedUser === '0') {
+      setOnSubmit(true);
+    } else {
+      setOnSubmit(false);
+
+      const user = usersFromServer.find(u => u.id === Number(selectedUser));
+
+      setTodos(prevTodos => {
+        const maxId = Math.max(...prevTodos.map(todo => todo.id), 0);
+        const newTodo = {
+          id: maxId + 1,
+          title: title,
+          completed: false,
+          userId: Number(selectedUser),
+          name: user?.name || '',
+          username: user?.username || '',
+          email: user?.email || '',
+        };
+
+        return [...prevTodos, newTodo];
+      });
+      setTitle('');
+      setSelectedUser('0');
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form onSubmit={handleOnSubmit} action="/api/todos" method="POST">
```

Comment:
The `action` and `method` attributes in the form tag are unnecessary since the form submission is handled by React. Consider removing these attributes to avoid confusion.